File: Combination.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Combination{
	static void Main(string [] args){
		int n=4;
		int k=2;
		Solution s = new Solution();
		var res = s.printPattern(n,k);
		Console.WriteLine(res[0][0]);

	}
}

class Solution{
	public IList<IList<int>> printPattern(int n,int k){
		IList<IList<int>> res=new List<IList<int>> ();
		IList<int> temp = new List<int>();
		makePattern(res,temp,n,1,k);
        return res;

	}
	public void makePattern(IList<IList<int>> res,IList<int> temp,int n,int left,int k){
		if(k==0){
			res.Add(temp);
			Console.Write(res[0][0]);
			return ;
		}
		for(int i=left;i<=n;++i){
			temp.Add(i);
			makePattern(res,temp,n,i+1,k-1);
			temp.RemoveAt(temp.Count-1);
		}
	}

}

File: Combination.cpp
#include <bits/stdc++.h>

using namespace std;

void makeCombinationHelper(vector<vector<int>> &res,vector<int> &temp,int left,int n,int k){
	if(k==0){
		res.push_back(temp);
		return;
	}
	for(int i=left;i<=n;i++){
		temp.push_back(i);
		makeCombinationHelper(res,temp,i+1,n,k-1);
		temp.pop_back();
	}
}
vector<vector<int>> makeCombi(int n,int k){
	vector<vector<int>> res;
	vector<int> temp;
	makeCombinationHelper(res,temp,1,n,k);
	return res;
}

int main()
{
	vector<vector<int>> res=makeCombi(4,2);
	cout<<res[0][0];

	return 0;
}
